### ISSUE TITLE
Is this the same as IP-Adapter faceID?

### ISSUE THREAD

**Issue by mrT333** [InstantID/InstantID#3]
Is this the same as the recently released ( [IP-Adapter faceID](https://huggingface.co/h94/IP-Adapter-FaceID) ) with some extra features like identity interpolation or are there any substantial benefits of instantID that IP-Adapter faceID doesn't provide?

I was astonished by the quality of the results of this project on the InstantID project page and I'm greatly looking forward for the release of the training / finetuning and inference code as well as the weights of this project. Do you have any rough estimates when the release will happen? Is it days / weeks / months away?

I can't wait for the release of this exciting technological breakthrough but if it takes more than a few weeks I'd probably just start implementing and finetuning the recently released IP-Adapter faceID for my upcoming projects.

**Comment by imperator-maximus** [comment]
I hope that this is not the same because you are not allowed to use faceID for commercial projects (so for example your upcoming projects have to be non-commercial). 

The Apache license used this project here looks promising here already.

**Comment by mrT333** [comment]
> [...] (so for example your upcoming projects have to be non-commercial). [...]


Yes my projects are indeed non commercial (academic) so it is not a deal breaker for me. But I agree an apache2 license will be way more valuable to the open source community longterm than a non-commercial one even if both instantID and IP-Adapter FaceID turn out to be very similiar in feature and quality.

**Comment by antonioo-c** [comment]
Hello @mrT333 , thank you for your appreciation of our work, I am a member of the InstantID team. I will answer some of your questions for our team.

1. We have noticed the recent progress of IP-adapter FaceID, but we have not tried it out yet. As shown in our project page, InstantID is capable of _Stylized Synthesis, Unusual Facial Stylized Synthesis, Photorealism Synthesis, Non-portrait Stylized Synthesis, Novel View Synthesis, Identity interpolation, Multi-ID Synthesis_. We are not sure whether the faceID can achieve all these features with ID consistency.
2. We are working very hard on releasing the paper and code, but it will still take some time before it's done. The estimated release date would be in March 2024.

**Comment by mrT333** [comment]
@antonioo-c thank you for the estimation that helps immensely. 

One of my focus areas is Novel View Synthesis and at least from the provided examples IPadapter FaceID might be lacking here as the faces tend to stick too close to the reference image regarding viewing angle. I have yet to try if forcing novel views (e.g. through controlnet) destroys the fidelity/ID consistency when using IPadapter FaceID.

I'm still greatly looking forward to the release of instantID, especially due to the Apache 2 license, not for my (non-commercial) projects but for the greater OS community impact it can have giving more incentive for numerous downstream projects and further development of the technology.

And if InstantID turns out to have much greater or more robust fidelity / ID consistency regarding Novel View Sythesis I probably switch back from faceID anyway when your project releases.

**Comment by haofanwang** [comment]
Thanks for all your interests. I'm also one the authors of this project. As @antonioo-c has suggested, we do have plan to release code, so, no worries. However, the code needs to go through some internal approvals before it is open source.

The good news is that we expect to release the first draft of the paper within January (the original plan was this month, but it was postponed due to the Christmas holiday), and we will provide an online demo for everyone to try. In fact, our model structure is quite easy, at that time, we also encourage everyone to upload unofficial model weights.

**Comment by ResearcherXman** [comment]
Thanks for all your interests. As @antonioo-c has suggested, we do have plan to release code, so, no worries. However, the code needs to go through some internal approvals before it is open source. Furthermore, the project license is temporary, we will finalize this when the code is released.

The good news is that we expect to release the first draft of the paper within January (the original plan was this month, but it was postponed due to the Christmas holiday), and we will provide an online demo for everyone to try. In fact, our model structure is quite easy, we encourage everyone to reproduce and upload their unofficial model weights.

By the way, we have indeed noticed that many similar projects (IP-Adapter with FaceID, FaceStudio, PhotoMaker and etc) have been released recently. But currently we are quite confident that our results far exceed theirs, so thank you for
patience and love you all. Merry Xmas and Happy New Years!

**Comment by mrT333** [comment]
Happy to hear the release of the paper is coming soon. I'm looking forward to build on top of InstantID in future downstream projects if the license allows for that (my current upcoming projects are non-commercial but some of the later ones also include commercial usage)

Is commercial use with the upcoming license still possible? 

ID consistent technology like this will be an essential building block in the currently emerging generative AI film-making industry, as characters need to stay consistent from scene to scene to allow for longer animation and movies or else the immersion of the viewer is broken. The identity of the character needs to be preserved in order for the viewer to get invested in them and their story.

In such emerging markets small businesses and indie-studios will be the first to establish themselves as they are more risk taking than bigger studios. 

If you intend to switch to a non-commercial license on the final release I hope you can consider to include at least a limited commercial use exemption (similar to the [LLAMA 2 license by Meta](https://ai.meta.com/resources/models-and-libraries/llama-downloads/) ) where small startups below X users or X monthly revenue are free to use but beyond that would need to pay license fees.

This would allow first movers like the mentioned indie-studios and small startups to first establish themselves and grow the gen-AI film making industry and pay back later once they grew in size and the ecosystem and community is already flourishing.

Looking forward to your answer and I wish you both @antonioo-c  and @ResearcherXman a Merry Xmas and Happy New Years as well!
